### ISSUE TITLE
Option to read TripStatistics (aggregated longTerm & tripStatistics)

### ISSUE THREAD

**Issue by msome** [tillsteinbach/WeConnect-python#13]
Would be great to read out the trip statistics which are displayed in the VW web frontend in various diagrams, see picture.
It allows to display nice statistics/graphs using HighCharts - about electric consumption, comparing winter & summer for auxConsumption during trips & charging habits.

Due to the huge tables which are behind trip statistics, a json dump of this element would be easier to handle than many many single fields which must be queried one-by-one. But does not matter, any way of delivering the table data is fine. 

Alternative simple solution not found, an ioBroker plugin allows such readings (longTerm or shortTerm) but unfortunately not both types at the same time and installing ioBroker for just this usage is like breaking a fly on the wheel (de: mit Kanonen auf Spatzen schießen).

Note: In the old api, the trips were located in rtsViewModel.longTermData & rtsViewModel.tripStatistics. For the new API (is it completely new?) I cannot tell.
![image](https://user-images.githubusercontent.com/25430864/126066365-6043b5fa-67e0-49ab-9fda-aa3ac29208b1.png)

Thank you for your time and consideration.

P.S. If you need test-data from api created for an e-Up, also independent from this ticket, feel free to contact me.

**Comment by tillsteinbach** [comment]
Hey, unfortunately the new API does not (yet) provide the trips as it focuses on ID drivers and the cars do not provide the data with the current firmware. But let me assure you: the day the data is provided on the API I'll start the implementation. I'm also eager to have this as it also provides the odometer.

**Comment by mpschr** [comment]
Is this still unsupported or has the API changed recently to support the retrieval of these statistics?

**Comment by tillsteinbach** [comment]
Yes, so far not supported.

**Comment by mpschr** [comment]
Ok, thanks for your feedback. Where would I look up if the API changes?

**Comment by tillsteinbach** [comment]
That is not so easy, but the most obvious would be that the Trips become available in the WeConnect ID App

**Comment by Mavyre** [comment]
What about using the "old" WeConnect API? It would also enhance the hybrid cars compatibility. Afaik, the "new" WeConnect ID app have issues with charging locations and climate control on GTEs

**Comment by tillsteinbach** [comment]
At least climatization works on my GTE flawlessly.
In general there would be a way to also use the old API, but I don't have the time to maintain the code for both at the same time. Also I believe that they will drop the support for the old one some time in the future and only use the new app. Otherwise there would be no reason making the old cars available in the new API

**Comment by Mavyre** [comment]
Sure, you're right that climate control is working flawlessly here too (Golf GTE 7 of 2020). But things like Hazard light flashes, Hazard + Honk, Window heating, Departure Times (despite an icon being in the new ID app), and of course, trip statistics are missing in the new app. Maybe the 2021+ cars are working a lot better on the new app...
I understand that maintaining the two different APIs from the apps is difficult. However, if you'd want some help, I can do some preliminary research and reverse engineer on the API from the "old" Weconnect app and check if it would be difficult to integrate and maintain.

**Comment by Mavyre** [comment]
I just used the CLI, and I can confirm that window heating is available on the new API, just not displayed on the new ID app (surprisingly). Departure Timers are, however, not available. Let's wait & see when they will switch everybody to WeConnect ID

**Comment by tillsteinbach** [comment]
On my 2021 Passat GTE I have departure timers:

```
[departureTimers] Elements: 1 items
        [departureTimersStatus] (last captured 2021-07-10T15:05:13+00:00)
                Heater Source: electric
                Min SoC: 50%
                Timers: 3 items
                        1: Enabled: False, Climatisation: True, Charging: False at 07:30 on monday tuesday wednesday thursday friday
                        2: Enabled: False, Climatisation: True, Charging: False at 2021-12-26T03:10:00
                        3: Enabled: False, Climatisation: False, Charging: True at 12:00 on monday tuesday wednesday thursday friday saturday sunday
                                Preferred charging times: 1 items
                                        1: Enabled: False, Start: 00:00, End: 00:00
```

or are you missing the possibility to add timers? That is something on my todo list, but I have to come up with a good API for that.

**Comment by Mavyre** [comment]
I talked too fast. I do have it too:
```
2: Enabled: False, Climatisation: True, Charging: True at 2022-10-17T13:59:00
        Preferred charging times: 1 items
                1: Enabled: True, Start: 19:59, End: 03:59
3: Enabled: False, Climatisation: True, Charging: True at 2022-10-18T06:15:00
        Preferred charging times: 1 items
                1: Enabled: True, Start: 20:29, End: 03:59
1: Enabled: True, Climatisation: True, Charging: True at 01:59 on monday tuesday wednesday thursday friday saturday sunday
```

It's just missing from the Weconnect ID app. So indeed WeConnect ID API does have everything (except the trip statistics and hazard light control)...

**Comment by tillsteinbach** [comment]
Update: there is now a new endpoint for trip data, unfortunately it still returns an error, but we are closer than ever to this!

**Comment by tillsteinbach** [comment]
Available now in release 0.56.0 for cars that support trip statistics (tested with Passat GTE, works; ID.3 SW 3.0, does not work)
